.github/workflows/go.yml

```diff
@@ -142,45 +141,70 @@ jobs:
-      run: |
-        version=2.3.1 # latest stable version
-        arch=amd64
-        curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${version}/kubebuilder_${version}_linux_${arch}.tar.gz"
-        tar -zxvf kubebuilder_${version}_linux_${arch}.tar.gz
-        mv kubebuilder_${version}_linux_${arch} kubebuilder && sudo mv kubebuilder /usr/local/
-        export PATH=$PATH:/usr/local/kubebuilder/bin
-    - name: Get dependencies
-      run: |
-        go get -u github.com/ory/go-acc
-    - name: install tray-agent dependencies
-      run: sudo apt update && sudo apt-get install gcc libgtk-3-dev libappindicator3-dev libwebkit2gtk-4.0-dev xclip
-    - run: go-acc ./... --ignore liqo/test/e2e
-    - name: Send coverage
-      uses: shogo82148/actions-goveralls@v1
-      with:
-        path-to-profile: coverage.txt
+      - name: Set the commit to build
+        run: |
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]; then
+             echo "commit_ref=${{ github.event.pull_request.head.sha }}" >> $GITHUB_ENV
+          elif [ ! -z ${{ steps.get_version.outputs.VERSION }} ]; then
+             echo "commit_ref=${{ steps.get_version.outputs.VERSION }}" >> $GITHUB_ENV
+          else
+             echo "commit_ref=${{ github.sha }}" >> $GITHUB_ENV
+          fi
+      - name: Set up Go 1.14
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
+        id: go
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          ref: "${{ env.commit_ref }}"
+      - name: Install Kubebuilder
+        run: |
+          version=2.3.1 # latest stable version
+          arch=amd64
+          curl -L -O "https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${version}/kubebuilder_${version}_linux_${arch}.tar.gz"
+          tar -zxvf kubebuilder_${version}_linux_${arch}.tar.gz
+          mv kubebuilder_${version}_linux_${arch} kubebuilder && sudo mv kubebuilder /usr/local/
+          export PATH=$PATH:/usr/local/kubebuilder/bin
+      - name: Get dependencies
+        run: |
+          go get -u github.com/ory/go-acc
+      - name: install tray-agent dependencies
+        run: |
+          export DEBIAN_FRONTEND=noninteractive
+          sudo apt update && sudo apt -y upgrade
+          sudo apt-get -y install gcc libgtk-3-dev libappindicator3-dev libxapp-dev libwebkit2gtk-4.0-dev xclip
+      - run: go-acc ./... --ignore liqo/test/e2e
+      - name: Send coverage
+        uses: shogo82148/actions-goveralls@v1
+        with:
+          path-to-profile: coverage.txt
+      - name: Build Agent asset
+        run: |
+          GO111MODULE=on CGO_ENABLED=1 GOOS=linux GOARCH=amd64 go build ./cmd/tray-agent/liqo-agent.go
+          tar -czf liqo-agent.tar.gz liqo-agent
+        if: github.event_name == 'push' && github.event.repository.full_name == 'liqotech/liqo' && startsWith(github.ref, 'refs/tags/v')
+      - name: Create Release
```

Comment:
May be we can cache the artifact and retrieve it in the "Release" job, where the release should be created ?

https://docs.github.com/en/free-pro-team@latest/actions/guides/storing-workflow-data-as-artifacts

Reply:
yes. I explored this possibility and chose to continue on the same workflow as there are just two additional steps, but we can move them.

Reply:
Thanks